hapi/hedera-protobufs/services/custom_fees.proto

```diff
@@ -173,3 +173,59 @@ message AssessedCustomFee {
    */
   repeated AccountID effective_payer_account_id = 4;
 }
+
+/**
+ * A custom fee definition for a consensus topic.
+ *
+ * This fee definition is specific to an Hedera Consensus Service (HCS) topic
+ * and SHOULD NOT be used in any other context.<br/>
+ * All fields for this message are REQUIRED.<br/>
+ * Only "fixed" fee definitions are supported because there is no basis for
+ * a fractional fee on a consensus submit transaction.
+ */
+message ConsensusCustomFee {
+  /**
+   * A fixed custom fee.
+   * <p>
+   * The amount of HBAR or other token described by this `FixedFee` SHALL
+   * be charged to the transction payer for each message submitted to a
+   * topic that assigns this consensus custom fee.
+   */
+  FixedFee fixed_fee = 1;
+
+  /**
+   * A collection account identifier.
+   * <p>
+   * All amounts collected for this consensus custom fee SHALL be transferred
+   * to the account identified by this field.
+   */
+  AccountID fee_collector_account_id = 2;
+}
+
+/**
+ * A wrapper for consensus custom fee list.
+ * <p>
+ * It is used in consensus_update_topic and it helps to differentiate between
+ * setting an empty list and not updating the list.
+ */
+message ConsensusCustomFeeList {
+  /**
+   * A set of custom fee definitions.<br/>
+   * These are fees to be assessed for each submit to this topic.
+   */
+  repeated ConsensusCustomFee fees = 1;
+}
+
+/**
+ * A wrapper for fee exempt key list.
+ * <p>
+ * It is used in consensus_update_topic and it helps to differentiate between
+ * setting an empty list and not updating the list.
```

Comment:
We are writing a specification, not a developer description here.
```suggestion
 * A wrapper for fee exempt key list.<br/>
 * This wrapper exists to enable an update transaction to differentiate between
 * a field that is not set and an empty list of keys.
 *
 * An _unset_ field of this type SHALL NOT modify existing values.<br/>
 * A _set_ field of this type with an empty `keys` list SHALL remove any
 * existing values.
```

Reply:
Fixed